### PR TITLE
Rearrange SIMD parts

### PR DESCRIPTION
**Pull request by lamphamsy**
- Rename `simd_fnt.h` to `simd_radix2_fft.h` includes operations dedicated for radix-2 FFT
- Move accelerated operations dedicated for RingModN from `simd_basic.h` to a new file `simd_ring.h`
- Rename `simd_basic.h` to `simd_fnt.h` includes operations dedicated for Fermat GF(q)

### CODE REVIEW

File: src/simd_basic_fnt.h [hunk 1]
```diff
@@ -28,8 +28,8 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
```
Comment (slaperche-scality) [inline src/simd_basic_fnt.h]: Why `simd_basic_fnt` instead of simply `simd_fnt`?

File: src/simd_basic_fnt.h [hunk 2]
```diff
@@ -198,135 +198,6 @@ inline void add_props(
     }
 }
 
-/* ==================== Operations for RingModN =================== */
```
Comment (slaperche-scality) [inline src/simd_basic_fnt.h]: This commit could be merged with ` [New] simd_ring.h includes accelerated operations dedicated for RingModN ` no?
You are onyl moving code, not modifying it right?

If yes, it's clearer to do the move in a single commit.

File: src/simd.h [hunk 3]
```diff
@@ -60,10 +60,13 @@ namespace simd {
 #endif
 
 // Include basic operations
-#include "simd_basic.h"
+#include "simd_basic_fnt.h"
```
Comment (slaperche-scality) [inline src/simd.h]: Comment above should be updated to `Include FNT operations.` or `Include accelerated operations for FNT.`

File: src/simd_basic_fnt.h (hunk 2 above)
Comment (lamphamsy) [reply src/simd_basic_fnt.h]: yes, okay. I group commits concerning splitting `simd_basic` to `simd_fnt` and `simd_ring` into one commit.

**Pull request merged**